src/Controller/Consent.php

```diff
@@ -22,25 +22,30 @@ class Consent
 {
     public const DEMO_CONSENT_URI = 'https://account-d.docusign.com/oauth/auth';
     public const CONSENT_URI = 'https://account.docusign.com/oauth/auth';
+    public const GRANT_TYPE_AUTHORIZATION_CODE = 'authorization_code';
+    public const GRANT_TYPE_IMPLICIT = 'implicit';
 
     private $envelopeBuilder;
     private $router;
     private $consentUri;
     private $integrationKey;
+    private $grantType;
 
-    public function __construct(EnvelopeBuilderInterface $envelopeBuilder, RouterInterface $router, string $consentUri, string $integrationKey)
+    public function __construct(EnvelopeBuilderInterface $envelopeBuilder, RouterInterface $router, string $consentUri, string $integrationKey, string $grantType)
     {
         $this->envelopeBuilder = $envelopeBuilder;
         $this->router = $router;
         $this->consentUri = $consentUri;
         $this->integrationKey = $integrationKey;
+        $this->grantType = $grantType;
```

Comment:
And in the constructor: 

```suggestion
        if (!isset(self::REPONSE_TYPES[$responseType])) {
            throw new PutYourExceptionHere();
        }
        
        $this->responseType = self::REPONSE_TYPES[$responseType];
```